extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/PrometheusConfiguration.java

```diff
@@ -0,0 +1,65 @@
+package io.quarkus.kubernetes.deployment;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot
+public class PrometheusConfiguration {
```

Comment:
I'm kinda on the fence about having such a separate configuration root.
I think it might be weird for a used that prometheus configuration would only affect Kubernetes manifests.

@iocanel WDYT?

Reply:
This is what I was asking about. I ended up making it its own thing because of how the annotations are added (as a subcontext for minikube, openshift, and vanilla kubernetes). 

I wanted to be able to re-ingest that prefix value.. 
    `@ConfigItem(defaultValue = "${quarkus.prometheus.prefix}/path")`

As soon as this configuration is nested, e.g. underneath Kubernetes: 
    @ConfigItem(defaultValue = "${quarkus.kubernetes.prometheus.prefix}/path")

It may be just how I attempted it the first time, but the hierarchy of configuration between the kubernetes types made this a little tricky for me to sort out how to do (specifically to ensure I could reuse that prefix as the default value for other attributes).

@geoand .. you're supposed to be reading. TOMORROW.

Reply:
I think we should probably move this configuration part of `io.quarkus.kubernetes.deployment.PlatformConfiguration`

Reply:
T -- that's where I put it initially, but I couldn't figure out how to do that w/o having it become `${quarkus.kubernetes.prometheus.prefix}/path` or `${quarkus.openshift.prometheus.prefix}/path` or ...

Reply:
It would be come that. But what is that a problem?

Reply:
See my first comment about setting default values.  I don't see how I can do that from one common place when the String is different depending on kube flavor.

Reply:
OK, I see.

You'll have to do that manually in the processor code.

Reply:
Right. From a documentation perspective (understanding how default values are set, etc), that's why I thought it was more appropriate (in the end) to make it its own thing. There are a few other cross-config references w/in the kube space, but they are into other config roots (container config), so this isn't a problem in those cases.

Reply:
Yeah, as this config is only about Kubernetes and not standalone Prometeus, it really sure should be properly namespaced.
It's a bit more work, but it should be worth it.